core/src/org/pentaho/platform/engine/core/system/PentahoSystem.java

```diff
@@ -225,133 +230,44 @@ public static boolean init( final IApplicationContext pApplicationContext ) {
-
-    if ( debug ) {
-      Logger.debug( PentahoSystem.class, "Setting property path" ); //$NON-NLS-1$
-    }
-    System.setProperty( "pentaho.solutionpath", "solution:" ); //$NON-NLS-1$
-    if ( LocaleHelper.getLocale() == null ) {
-      LocaleHelper.setLocale( Locale.getDefault() );
-    }
-
-    if ( PentahoSystem.systemSettingsService != null ) {
-      if ( debug ) {
-        Logger.debug( PentahoSystem.class, "Reading ACL list from pentaho.xml" ); //$NON-NLS-1$
-      }
-      // Set Up ACL File Extensions by reading pentaho.xml for acl-files
-      //
-      // Read the files that are permitted to have ACLs on them from
-      // the pentaho.xml.
-      //
-      String aclFiles = PentahoSystem.getSystemSetting( "acl-files", "xaction,url" ); //$NON-NLS-1$ //$NON-NLS-2$
-      StringTokenizer st = new StringTokenizer( aclFiles, "," ); //$NON-NLS-1$
-      String extn;
-      while ( st.hasMoreElements() ) {
-        extn = st.nextToken();
-        if ( !extn.startsWith( "." ) ) { //$NON-NLS-1$
-          extn = "." + extn; //$NON-NLS-1$
-        }
-        PentahoSystem.ACLFileExtensionList.add( extn );
-      }
-    }
-
-    if ( debug ) {
-      Logger.debug( PentahoSystem.class, "Set Java System Properties" ); //$NON-NLS-1$
-    }
-    PentahoSystem.setSystemProperties();
-
-    if ( debug ) {
-      Logger.debug( PentahoSystem.class, "Initialize XML Factories" ); //$NON-NLS-1$
-    }
-    PentahoSystem.initXMLFactories();
-
-    if ( debug ) {
-      Logger.debug( PentahoSystem.class, "Set Logging Level from pentaho.xml setting" ); //$NON-NLS-1$
-    }
-    PentahoSystem.loggingLevel = ILogger.ERROR;
-    if ( PentahoSystem.systemSettingsService != null ) {
-      PentahoSystem.loggingLevel =
-          Logger.getLogLevel( PentahoSystem.systemSettingsService.getSystemSetting( "log-level", "ERROR" ) ); //$NON-NLS-1$//$NON-NLS-2$
-    }
-
-    Logger.setLogLevel( PentahoSystem.loggingLevel );
-
-    // to guarantee hostnames in SSL mode are not being spoofed
-    if ( debug ) {
-      Logger.debug( PentahoSystem.class, "Register host name verifier" ); //$NON-NLS-1$
-    }
-    PentahoSystem.registerHostnameVerifier();
-
-    assert null != aggObjectFactory : "aggObjectFactory must be non-null"; //$NON-NLS-1$
-    try {
-      if ( debug ) {
-        Logger.debug( PentahoSystem.class, "Validating object factory" ); //$NON-NLS-1$
-      }
-      PentahoSystem.validateObjectFactory();
-    } catch ( PentahoSystemException e1 ) {
-      throw new RuntimeException( e1 ); // this is fatal
-    }
-
-    // store a list of the system listeners
-    try {
-      if ( debug ) {
-        Logger.debug( PentahoSystem.class, "Start System Listeners" ); //$NON-NLS-1$
+    } else {
+      PentahoSystem.applicationContext = pApplicationContext;
+      PentahoSystem.initializedStatus = PentahoSystem.SYSTEM_INITIALIZED_OK;
+
+      if ( useThread ) {
+        final Thread startingThread = new Thread( new Runnable() {
```

Comment:
@pentaho-nbaker @lucboudreau can you tell why should we make the thread a daemon? Can't find any reasonable reason.